appveyor.yml

```diff
@@ -0,0 +1,16 @@
+platform:
+    - x64
+
+cache:
+    - C:\Users\appveyor\.esy\3_\i
+    - C:\Users\appveyor\.esy\source-tarballs
+
+install:
+    - npm install -g esy@0.3.0
+    - appveyor-retry esy install
+
+build_script:
+    - esy build
+    - esy b make test
```

Comment:
this is not truly using `esy` and its sandbox because `make test` is still building with dune. Additionally it's building the oprint test binary (should perhaps also be built with Esy too?)

Reply:
> this is not truly using esy and its sandbox because make test is still building with dune. 

Hmm, I'm not sure what you mean here - `esy b` means run a command in the esy build environment. It's often the case we'll run dune in the build environment via `esy`!😄 

But unfortunately this isn't sufficient to run the tests on Windows - here's the error log from AppVeyor: 
```
node ./formatTest/testOprint.js
bash: node: command not found
```
(`node` isn't present in the esy sandbox, so it fails)

In any case, I'll leave this out for now and track it separately - there's more work to get the tests running. 

Moving the oprint test binary outside of the Makefile as you suggested would be a good start; we'd also need to refactor the bash-driven tests to something that would work well in Windows too. Looks like even the node tests make some assumptions about being on a POSIX style platform (ie, they call `cat` which isn't around by default on Windows)

Reply:
@bryphe if you rebase on master it should work! I recently eliminated the need for Node.js in https://github.com/facebook/reason/pull/2213

Reply:
Ah, with respect to the other POSIX problems I can't help.

Reply:
Ah cool, thanks @anmonteiro ! I'll try it out - would be great to have some tests running on Windows.

Reply:
More details here on test blockers: #2224 . Thanks for looking the change over @anmonteiro ! 👍